src/XrdOss/XrdOssRename.cc

```diff
@@ -110,12 +110,11 @@ int XrdOssSys::Rename(const char *oldname, const char *newname,
      && (((retc = GenRemotePath(oldname, remote_path_Old))
      ||   (retc = GenRemotePath(newname, remote_path_New)))) ) return retc;
 
-// Make sure that the target file does not exist
+// If on a remote FS or symlink, make sure that the target path does not exist
 //
-#if 0
-   retc2 = lstat(local_path_New, &statbuff);
-   if (!retc2) return -EEXIST;
-#endif
+   if (!(retc2 = lstat(local_path_New, &statbuff)))
+      { if (remotefs || (statbuff.st_mode & S_IFMT) == S_IFLNK) return -EEXIST;
+      }
```

Comment:
What this means is that any symlinked FS will not allow a rename on top of an existing link. Is that what you really want? That excludes any virtualized filesystem even if local.

Reply:
The symlink part comes from your original comment. In my case I am mostly interested to local file, no symlinks. However, I see that the man page of rename says that

if newpath refers to a symbolic link, the link will be overwritten

So I suggest that we only leave the check on 'remotefs'. What do you think?